frontend/src/scenes/notebooks/Suggestions/FloatingSuggestions.tsx

```diff
@@ -1,52 +1,66 @@
+import { useEffect, useState } from 'react'
 import { notebookLogic } from '../Notebook/notebookLogic'
+import { useResizeObserver } from 'lib/hooks/useResizeObserver'
 
 export function FloatingSuggestions({ editor }: { editor: TTEditor }): JSX.Element | null {
     const logic = insertionSuggestionsLogic()
     const { activeSuggestion, previousNode } = useValues(logic)
     const { setEditor } = useActions(logic)
     const { editor: notebookEditor } = useValues(notebookLogic)
+    const { ref: setRef, height } = useResizeObserver()
+    const [shouldShow, setShouldShow] = useState<boolean>(false)
+
+    const [position, setPosition] = useState<{ top: number }>({ top: 0 })
 
     const { Component } = activeSuggestion
 
+    const handleUpdate = (): void => {
+        const selection = window.getSelection()
+
+        if (selection && selection.anchorNode && selection.anchorNode.parentElement) {
+            if (selection.anchorNode.nodeType === Node.ELEMENT_NODE) {
+                const editorPos = editor.view.dom.getBoundingClientRect()
+                const selectionPos = (selection.anchorNode as HTMLElement).getBoundingClientRect()
+
+                setPosition({ top: selectionPos.top - editorPos.top })
+            }
+        }
+
+        setShouldShow(
+            editor.view.hasFocus() && editor.isEditable && editor.isActive('paragraph') && isCurrentNodeEmpty(editor)
+        )
+    }
+
     useEffect(() => {
         setEditor(notebookEditor)
     }, [notebookEditor])
 
+    useEffect(() => {
+        handleUpdate()
+    }, [height])
+
+    useEffect(() => {
+        editor.on('update', handleUpdate)
+        editor.on('selectionUpdate', handleUpdate)
+        setRef(editor.view.dom)
+        return () => {
+            editor.off('update', handleUpdate)
+            editor.off('selectionUpdate', handleUpdate)
+        }
+    }, [])
+
     return (
-        <FloatingMenu
-            editor={editor}
-            tippyOptions={{
-                duration: [100, 0],
-                placement: 'right',
-                offset: [0, 0],
-            }}
-            className="NotebookFloatingButton"
-            shouldShow={({ editor }: { editor: TTEditor }) => {
-                if (!editor) {
-                    return false
-                }
-                if (
-                    editor.view.hasFocus() &&
-                    editor.isEditable &&
-                    editor.isActive('paragraph') &&
-                    isCurrentNodeEmpty(editor)
-                ) {
-                    return true
-                }
-
-                return false
-            }}
-        >
-            <div className="FloatingSuggestion flex items-center justify-content">
-                {Component && <Component previousNode={previousNode} editor={notebookEditor} />}
-            </div>
-        </FloatingMenu>
+        <div className="NotebookFloatingButton absolute" style={{ zIndex: 9999, top: position.top, left: 0 }}>
```

Comment:
I'm nit-picking but the zIndex should be a proper var, maybe the same as the popover z-index

Reply:
No, nit picking is good. I'd hate if I was someone else who came across it in future

Reply:
Fixed this up. Anything else you want to see changed?